### PR TITLE
samples: nrf9160: modem_shell: threading support for ping

### PR DESCRIPTION
**Pull request by jhirsi**
Added a support for the ping command to be running in a separate thread within a th -command.
Note: due to limitations in raw sckt that ping is using, ping cannot be run simultaneously with other threads/commands, within the LTE connection with same PDN ID.

Additionally, added some more instructions for smoother pppd Linux usage and also removed iperf3 from Thingy builds in order to fit into a flash.

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -175,6 +175,7 @@ nrf9160 samples
         AT command termination methods can be configured using Kconfig options.
         The default method, is CR termination.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:176]:
```suggestion
        The default method is CR termination.
```
Sorry, this part was not really subject to review, but I happened to see an extra comma here. I missed it myself in the previous review.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -175,6 +175,7 @@ nrf9160 samples
...
         In AT command mode, a maximum of 10 AT commands can be pipelined with ``|`` as the delimiter character between pipelined AT commands.
+      * Threading support for ping command.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:178]:
```suggestion
      * Threading support for the ``ping`` command.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (jhirsi) [reply doc/nrf/releases/release-notes-changelog.rst]: Changed

**Review by peknis — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/th/th_shell.c [hunk 3]
```diff
@@ -79,13 +79,19 @@ static int cmd_th(const struct shell *shell, size_t argc, char **argv)
...
 	SHELL_CMD_ARG(
 		results, NULL,
 		"th results <thread nbr>\nGet results for the background thread.",
 		cmd_th_get_results, 2, 0),
 	SHELL_CMD_ARG(kill, NULL, "th kill <thread nbr>\nKill a thread.", cmd_th_ctrl_kill, 2, 0),
 	SHELL_CMD_ARG(status, NULL, "Get status of the threads.", cmd_th_status, 1, 0),
 	SHELL_SUBCMD_SET_END);
-SHELL_CMD_REGISTER(th, &sub_bg, "Commands for running iperf3 in separate threads.", cmd_th);
+SHELL_CMD_REGISTER(th, &sub_bg,
+		   "Commands for running iperf3 and in separate threads.\n"
```
Comment (rlubos) [inline samples/nrf9160/modem_shell/src/th/th_shell.c:94]: Shouldn't this be `iperf3 and ping`?

Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/th/th_shell.c]: yes, fixed.

File: samples/nrf9160/modem_shell/src/th/th_ctrl.c [hunk 4]
```diff
@@ -264,11 +291,24 @@ static void th_ctrl_data_start(struct th_ctrl_data *data,
...
+#if defined(CONFIG_MOSH_PING)
+	if (!strcmp(argv[1], "ping")) {
+		start = true;
+	}
+#endif
+	if (!start) {
+		mosh_error("Cannot start a thread: command %s", argv[1]);
 		return;
 	}
+	/* TODO: there cannot be iperf3 and ping running at same time at the same context */
```
Comment (tokangas) [inline samples/nrf9160/modem_shell/src/th/th_ctrl.c:311]: Is this comment here intentionally or should this check be added to this PR?

Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/th/th_ctrl.c:311]: Implementation time decision was that no check for that is done, limitations are noted in related shell cmd help. Let that be enough. I'll remove this one.

Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/th/th_ctrl.c]: Comment was removed.

**Review by tokangas — APPROVED** [review]

**Pull request merged**